irobotclient/configuration_handler.py

```diff
@@ -0,0 +1,159 @@
+    if args.input_file.endswith('/') or os.path.isdir(args.input_file):
+        raise IrobotClientException(errno=errno.ECONNABORTED,
+                                    message="Cannot download entire directories at present.")
+
+    if args.input_file.startswith('/'):
+        args.input_file = args.input_file.lstrip('/')
+
+
+def _check_output_directory_argument(args):
+    """
+    Check if the output directory already exists and if it already contains files of the same name as the input file.
+
+    :return: n/a
+
+    """
+
+    try:
+        # Expand the output_dir argument so the full directory path can be used in the rest of the program.
+        args.output_dir = os.path.expanduser(args.output_dir)
+
+        # Add a trailing slash to indicate directory.
+        if not args.output_dir.endswith('/'):
+            args.output_dir = args.output_dir + '/'
+
+        # Split up the path string to obtain just the file name without extensions and full path.
+        file_name = os.path.splitext((os.path.split(args.input_file))[1])[0]
+
+        for dir_file in os.listdir(args.output_dir):
+
+            # Basic match; will return true if the directory file begins with the input_file string.
+            # TODO - Improve regex on checking whether input file already exists.
+            if re.match(os.path.splitext(dir_file)[0], file_name) and not args.force:
+                raise IrobotClientException(errno=errno.EEXIST, message="File already exists. Please use the "
+                                                                        "--force option to overwrite.")
+
+    except IrobotClientException:
+        raise
+    except OSError:
+        raise
+    except:
+        raise Exception("UNKNOWN ERROR: Failed to validate the output directory.")
+
+
+def _check_url_argument(args):
+    """
+    Check if a url has been provided via command line or environment setting and check trailing slash.
+
+    :param args:
+    :return:
+    """
+
+    try:
+        if args.url is None:
+            args.url = os.environ['IROBOTURL']
+    except KeyError:
+        raise IrobotClientException(errno.ENOKEY, "Cannot set URL from command line argument or environment"
+                                                  " variable.")
+    except:
+        raise Exception("UNKNOWN ERROR: Failed to set the URL.")
+
+    if not args.url.endswith('/'):
+        args.url += '/'
+
+
+def _check_authorisation_token(args):
+    """
+    Check if the authorisation token has been set on the command line or environment variable.  If no
+
+    :param args:
+    :return:
+    """
+
+    if args.token is None:
+        args.token = os.environ['ARVADOSTOKEN']
+
+
+def run():
+    """
+    Calls the functions to collect any command line arguments, set configuration details needed for the iRobot
+    requests, and return the argparse object to the request formatter.
+
+    :return:
+    """
+
+    args = _get_command_line_agrs()
+    _validate_command_line_args(args)
```

Comment:
Your authorisation argument parsing is fine, but iRobot always needs credentials, so it would be good -- if no credentials are found (either Basic or Arvados, from the command line or environment) -- that an exception is thrown. You don't need this, because iRobot will return a 401 and that'll presumably be handled by your client somewhere, but it's always good to bailout early if/when you can.